### PR TITLE
Spawn a Julia worker via the Raylet

### PR DESCRIPTION
**Pull request by omus**
Working off of @glennmoy's #2 PR. These changes result in `ray start` automatically spawning a Julia worker upon startup which waits to run tasks.

### CODE REVIEW

File: python/ray/_private/runtime_env/context.py [hunk 1]
```diff
@@ -64,12 +64,16 @@ def exec_worker(self, passthrough_args: List[str], language: Language):
             passthrough_args = class_path_args + passthrough_args
         elif language == Language.JULIA:
             executable = "julia"
+            julia_args = ["--project=/Users/cvogt/.julia/dev/ray_core_worker_julia_jll"]
```
Comment (omus) [inline python/ray/_private/runtime_env/context.py:67]: Just to get started but we'll need to figure out a better way

File: python/ray/_private/runtime_env/context.py [hunk 2]
```diff
@@ -84,7 +88,7 @@ def exec_worker(self, passthrough_args: List[str], language: Language):
                 + " "
                 + command_str
             )
-        logger.debug(f"Exec'ing worker with command: {command_str}")
+        logger.info(f"Exec'ing worker with command: {command_str}")
```
Comment (omus) [inline python/ray/_private/runtime_env/context.py:91]: Didn't look into where the log level was being set

File: python/ray/_private/workers/setup_worker.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import argparse
 import logging
+import time
```
Comment (omus) [inline python/ray/_private/workers/setup_worker.py:3]:
```suggestion
```

File: src/ray/common/function_descriptor.h [hunk 4]
```diff
@@ -358,6 +358,9 @@ inline bool operator==(const FunctionDescriptor &left, const FunctionDescriptor
   case ray::FunctionDescriptorType::kCppFunctionDescriptor:
     return static_cast<const CppFunctionDescriptor &>(*left) ==
            static_cast<const CppFunctionDescriptor &>(*right);
+  case ray::FunctionDescriptorType::kJuliaFunctionDescriptor:
+    return static_cast<const JuliaFunctionDescriptor &>(*left) ==
+           static_cast<const JuliaFunctionDescriptor &>(*right);
```
Comment (omus) [inline src/ray/common/function_descriptor.h:363]: Fixup for #1

File: src/ray/raylet/main.cc [hunk 5]
```diff
@@ -81,7 +81,7 @@ int main(int argc, char *argv[]) {
   InitShutdownRAII ray_log_shutdown_raii(ray::RayLog::StartRayLog,
                                          ray::RayLog::ShutDownRayLog,
                                          argv[0],
-                                         ray::RayLogLevel::INFO,
+                                         ray::RayLogLevel::DEBUG,
```
Comment (omus) [inline src/ray/raylet/main.cc:84]: Sets the logging level in the Raylet C++ code to show `DEBUG` and up messages

File: src/ray/raylet/worker_pool.cc [hunk 6]
```diff
@@ -162,6 +162,9 @@ void WorkerPool::Start() {
         "RayletWorkerPool.deadline_timer.kill_idle_workers");
   }
 
+  RAY_LOG(INFO) << "WUBBA";
+  PrestartDefaultCpuWorkers(Language::JULIA, 1);
```
Comment (omus) [inline src/ray/raylet/worker_pool.cc:166]: Starts the Julia worker upon `ray start --head`

File: python/ray/_private/runtime_env/context.py (hunk 1 above)
Comment (omus) [reply python/ray/_private/runtime_env/context.py]: I spent some time on this and we're now specifying the `JULIA_PROJECT` environmental variable to specify the project. This value is specified via the task specification from the Julia process which submitted the task. Overall this works great with the caveat that you cannot prestart workers unless the `ray_core_worker_julia_jll` package is installed globally

File: python/ray/_private/runtime_env/context.py [hunk 7]
```diff
@@ -42,9 +42,11 @@ def serialize(self) -> str:
 
     @staticmethod
     def deserialize(json_string):
+        logger.debug(f"deserialize: {json.loads(json_string)}")
```
Comment (kleinschmidt) [inline python/ray/_private/runtime_env/context.py:45]:
```suggestion
        # TODO(Beacon): remove these when we're done
        logger.debug(f"deserialize: {json.loads(json_string)}")
```

File: python/ray/_private/runtime_env/context.py [hunk 8]
```diff
@@ -42,9 +42,11 @@ def serialize(self) -> str:
...
         return RuntimeEnvContext(**json.loads(json_string))
 
     def exec_worker(self, passthrough_args: List[str], language: Language):
+        logger.debug(f"Worker context env: {self.env_vars}")
```
Comment (kleinschmidt) [inline python/ray/_private/runtime_env/context.py:49]:
```suggestion
        # TODO(Beacon): remove these when we're done      
        logger.debug(f"Worker context env: {self.env_vars}")
```

**Review by kleinschmidt — APPROVED** [review]
leaving a few breadcrumbs but otherwise looks good

**Pull request merged**